lib/modules/datasource/dotnet/index.ts

```diff
@@ -91,13 +93,15 @@ export class DotnetDatasource extends Datasource {
     if (body) {
       const type = DotnetDatasource.getType(packageName);
       const { releases: releases } = body;
-      result = releases.map((release) => {
-        return {
-          version: release[type].version,
-          releaseTimestamp: release['release-date'],
-          changelogUrl: release['release-notes'],
-        };
-      });
+      result = releases
+        .map((release) => {
+          return {
+            version: release[type]?.version,
+            releaseTimestamp: release['release-date'],
+            changelogUrl: release['release-notes'],
+          };
+        })
+        .filter((release) => is.nonEmptyString(release.version)) as Release[];
```

Comment:
why not filter before map? i think we don't need the cast then.

Reply:
The filter is a little more complex because I am checking the keys for null, but it works without a cast.